crates/nu-command/src/filesystem/mktemp.rs

```diff
@@ -0,0 +1,133 @@
+                "Optional pattern from which the name of the file or directory is derived. Must contain at least three 'X's in last component.",
+            )
+            .named("suffix", SyntaxShape::String, "Append suffix to template; must not contain a slash.", None)
+            .named("tmpdir-path", SyntaxShape::Filepath, "Interpret TEMPLATE relative to tmpdir-path. If tmpdir-path is not set use $TMPDIR", Some('p'))
+						.switch("tmpdir", "Interpret TEMPLATE relative to the system temporary directory.", Some('t'))
+            .switch("directory", "Create a directory instead of a file.", Some('d'))
+            .category(Category::FileSystem)
+    }
+
+    fn run(
+        &self,
+        engine_state: &EngineState,
+        stack: &mut Stack,
+        call: &Call,
+        _input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        let span = call.head;
+        let template: String = call
+            .rest(engine_state, stack, 0)?
+            .get(0)
+            .cloned()
+            .map(|i: Spanned<String>| i.item)
+            .unwrap_or("tmp.XXXXXXXXXX".to_string()); // same as default in coreutils
+
+        let directory = call.has_flag("directory");
+        let suffix: Option<String> = call
+            .get_flag(engine_state, stack, "suffix")?
+            .map(|i: Spanned<String>| i.item);
+        let tmpdir = call.has_flag("tmpdir");
+        let tmpdir_path: Option<PathBuf> = call
+            .get_flag(engine_state, stack, "tmpdir-path")?
+            .map(|i: Spanned<PathBuf>| i.item);
+
+        let tmpdir = if tmpdir_path.is_some() {
+            tmpdir_path
+        } else if directory || tmpdir {
+            Some(std::env::temp_dir())
+        } else {
+            Some(current_dir(engine_state, stack)?)
+        };
+
+        let options = uu_mktemp::Options {
+            directory,
+            dry_run: false,
+            quiet: false,
+            suffix,
+            template,
+            tmpdir,
+            treat_as_template: true,
+        };
+
+        let res = match uu_mktemp::mktemp(&options) {
+            Ok(res) => res
+                .into_os_string()
+                .into_string()
+                .map_err(|e| ShellError::IOErrorSpanned(e.to_string_lossy().to_string(), span))?,
+            Err(e) => {
+                return Err(ShellError::GenericError(
+                    format!("{}", e),
+                    format!("{}", e),
+                    None,
+                    None,
+                    Vec::new(),
+                ));
+            }
+        };
+        Ok(PipelineData::Value(Value::string(res, span), None))
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![
+            Example {
+                description: "Make a temporary file with the given suffix in the current working directory.",
+                example: "mktemp --suffix .txt",
+                result: Some(Value::string("<WORKING_DIR>/tmp.lekjbhelyx.txt", Span::new(0, 0))),
```

Comment:
```suggestion
                result: Some(Value::test_string("<WORKING_DIR>/tmp.lekjbhelyx.txt")),
```
We can use test_string() here since it already supplies and empty Span. Same with the others below.